### ISSUE TITLE
Math font error prevents the manual from building on NixOS

### ISSUE THREAD

**Issue by OlivierNicole** [sile-typesetter/sile#1552]
**OS type:** NixOS 22.05

After successfully building SILE (in a custom nix-shell with the right dependencies, I didn't use the Nix flake as I don't know how to use them yet), I attempted `make docs`:
```
...
[64] <./packages/autodoc/init.lua:154> as sil
[65] 
! You must use a math font for math rendering. at ./packages/autodoc/init.lua:154:30:26: in \mathml

error summary:
	Processing at: ./packages/autodoc/init.lua:154:30:26: in \mathml
	Using code at: ./packages/math/base-elements.lua:106: You must use a math font for math rendering.

Run with --traceback for more detailed trace leading up to errors.
make: *** [Makefile:1452 : documentation/sile.pdf] Erreur 1
make: *** Suppression du fichier « documentation/sile.pdf »
```
Yet Libertinus Math is installed, and is even present twice: as one of my system fonts, and it was downloaded by the makefile into `.fonts/`.

**Comment by alerque** [comment]
The flake is really easy to run. From anywhere *except* the SILE source directory (unless you aggressively clean it first) just run:

```console
# To get HEAD
$ nix run github:sile-typesetter/sile -- --version
SILE v0.14.3-1f42674-flake (Lua 5.3)

# To get any specific version
$ nix run github:sile-typesetter/sile/v0.14.3 -- --version
SILE v0.14.3-1e84ea3-flake (Lua 5.3)
```

But that wasn't your main question.

I'm not sure about what you bumped into honestly. I'm guessing something about the fontconfig in the nix shell is sandboxed in a way that kept the build system from finding either sets of fonts.

Were you building on the `shell.nix` from the source directory or something else? Do you have a shell file or something I could use to replicate your results?

Also what's the end goal here? With SILE being available in the Nix packages repo you should be able to just add `sile` as a dependency for a shell and be done with it, but I assume since you are trying to build the manual you must have development work on SILE in mind rather than *using* SILE in a project. In that case we probably should setup the repo that the necessary development environment can easily be run in strict mode.

**Comment by alerque** [comment]
> Exactly, here I was trying to hack into the manual (to see if I propose some aesthetic changes). I don't know if the flake is useful for development.

Yes, flake wouldn't be a good tool for that. It's great for other things, especially having access to ald versions really quickly for regression testing, but not for building the manual.

Rather than trying to copy relevant stuff from the derivation, did you try just using it directly?

```console
$ cd ~/projects/sile-typesetter/sile
$ git clean -dxff
$ nix develop nixpkgs#sile
bash-5.1$ ./bootstrap.sh
bash-5.1$ ./configure --with-system-luarocks
bash-5.1$ make
bash-5.1$ make docs
```

That works for me, but I'm also using `nix` on an Arch Linux system that has all the dependencies available if something fell through the `nix develop` environment (which I believe runs in impure mode).

**Comment by alerque** [comment]
Upon review of the documentation it looks like it `nix develop` can use either a derivation or flake, so you don't need to even bother with the *nixpkgs#sile* lookup. Just CD to the source directory and run `nix develop`. Forget the customized `shell.nix`.

Does that work for you like it does for me?

**Comment by OlivierNicole** [comment]
Thanks for the tip with `nix develop`. It works well, however I still have the same error when building the manual. Strangely, I can typeset the `tests/math-*` files without problem.

**Comment by alerque** [comment]
The issue is the Nix derivation does not specify `bsdtar` as a build dependency. It is not used to build SILE itself, but it is needed to build the docs, specifically to extract some of the downloaded fonts.

**Comment by alerque** [comment]
The extra requirement is the `libarchive` package. I'm not quite sure how to get it added to the `devShells` so that it ends up in `nix develop` but not required by just building the flake or derivation.

**Comment by OlivierNicole** [comment]
> The `.sources` directory has the archives, but when it tried to extract specific font files from them into `.fonts` it errors. You may have the file `.fonts/LibertinusMath-Regular.otf` but I'd be willing to bet it is 0 bytes.

Spot-on! I had seen the bsdtar errors but as `make` didn't stop, I thought they weren't fatal. I think that make should stop on those. (Not sure how to do that yet, might look later.)

**Comment by alerque** [comment]
I was (and am) puzzled why `make` does not bail out when it can't run something. I'll add checks for that tooling to `./configure`, but those recipes should abort too.

**Comment by alerque** [comment]
My mistake. I had set `.ONESHELL:` without also adding `.SHELLFLAGS += -e`.
